src/main/java/com/antondoddo/production/usecase/AddUseCase.java

```diff
@@ -0,0 +1,18 @@
+package com.antondoddo.production.usecase;
+
+import com.antondoddo.production.Production;
+import com.antondoddo.production.repository.Repository;
+import com.antondoddo.production.repository.exception.CouldNotAddProductionException;
+
+public class AddUseCase {
+
+  private Repository repository;
+
+  public AddUseCase(Repository repository) {
+    this.repository = repository;
+  }
+
+  public void execute(Production production) throws CouldNotAddProductionException {
```

Comment:
```suggestion
  public Production execute(AddDTO dto throws CouldNotAddProductionException {
```

Il metodo per l'add deve essere questo, cosi da ritornare la produzione aggunta a chi utilizzerà lo use case, ed al suo interno gestisci tutta la logica per creare una production 💪